yaml-tests/src/test/resources/uuid-proto.yamsql

```diff
@@ -0,0 +1,57 @@
+#
+# uuid-proto.yamsql
+#
+# This source file is part of the FoundationDB open source project
+#
+# Copyright 2021-2025 Apple Inc. and the FoundationDB project authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+setup:
+  connect: "jdbc:embed:/__SYS?schema=CATALOG"
+  steps:
+    - query: drop schema template if exists uuid_proto_template
+    - query: drop database if exists /FRL/UUID_PROTO_YAML
+    - query: create database /FRL/UUID_PROTO_YAML
+    - load schema template: UUID_PROTO_TEMPLATE from com.apple.foundationdb.relational.yamltests.generated.uuidtests.UuidProtoTestsProto
+    - query: create schema /FRL/UUID_PROTO_YAML/UUID_PROTO_SCHEMA with template UUID_PROTO_TEMPLATE
+    - set schema state: "{\"name\": \"UUID_PROTO_SCHEMA\", \"database_id\": \"/FRL/UUID_PROTO_YAML\", \"template_name\": \"UUID_PROTO_TEMPLATE\", \"store_info\" : {\"formatVersion\": 2}}"
+---
+setup:
+  connect: "jdbc:embed:/FRL/UUID_PROTO_YAML?schema=UUID_PROTO_SCHEMA"
+  steps:
+    - query: INSERT INTO T1(ID, COL1, COL2("most_significant_bits", "least_significant_bits")) VALUES
+        (1, 2, (11, 12)),
+        (3, 4, (13, 14)),
+        (5, 6, (15, 16)),
+        (7, 8, (17, 18));
+---
+test_block:
+  connect: "jdbc:embed:/FRL/UUID_PROTO_YAML?schema=UUID_PROTO_SCHEMA"
+  tests:
+    -
+      # test select *
+      - query: SELECT * from T1 where ID = 1;
+      - result: [{1, 2, {11, 12}}]
+    -
+      # test select fields
+      - query: SELECT COL1 from T1 where ID = 1;
```

Comment:
Having another test that returns multiple results but doesn't include the UUID would be good here, too. I think it should show that we get the expected results even as we turn UUID support on and off (and should work even in force continuations mode with `4.2.3.0` and `4.2.4.0`)